core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/validation/service/NifiUpgradeValidator.java

```diff
@@ -0,0 +1,67 @@
+package com.sequenceiq.cloudbreak.service.upgrade.validation.service;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cluster.api.ClusterApi;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateService;
+import com.sequenceiq.cloudbreak.common.exception.UpgradeValidationFailedException;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+
+@Component
+public class NifiUpgradeValidator implements ServiceUpgradeValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NifiUpgradeValidator.class);
+
+    private static final String NIFI_SERVICE_TYPE = "NIFI";
+
+    private static final String NIFI_WORKING_DIRECTORY = "nifi.working.directory";
+
+    private static final String NIFI_WORKING_DIRECTORY_VALUE = "/hadoopfs/";
```

Comment:
I would put this in `VolumeUtils`  and check with 
`nifiWorkingDirectory.get().startsWith(NIFI_WORKING_DIRECTORY_VALUE)` instead to be 100%